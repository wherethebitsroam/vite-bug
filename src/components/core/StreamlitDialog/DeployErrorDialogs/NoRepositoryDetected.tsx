import React from "react"
import { ONLINE_DOCS_URL, TEAMS_URL } from "src/urls"
import { IDeployErrorDialog } from "./types"

function NoRepositoryDetected(): IDeployErrorDialog {
  return {
    title: "Unable to deploy app",
    body: (
      <>
        <p>
          Could not find a remote repository hosted on GitHub. Are you sure you
          are on a branch that is tracking a remote GitHub branch?
        </p>
        <p>How Streamlit Cloud works:</p>
        <ul>
          <li>
            To deploy a public app, you must first put it in a public GitHub
            repo. See{" "}
            <a
              href={ONLINE_DOCS_URL}
              rel="noopener noreferrer"
              target="_blank"
            >
              our documentation
            </a>{" "}
            for more details.
          </li>
          <li>
            If you'd like to deploy a private app,{" "}
            <a href={TEAMS_URL} target="_blank" rel="noopener noreferrer">
              sign up for Streamlit for Teams
            </a>
            .
          </li>
        </ul>
      </>
    ),
  }
}

export default NoRepositoryDetected
