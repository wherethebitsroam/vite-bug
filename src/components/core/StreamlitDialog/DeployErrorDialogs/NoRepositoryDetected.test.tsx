import NoRepositoryDetected from "./NoRepositoryDetected"

describe("DeployErrorDialog - NoRepositoryDetected", () => {
  it("should render without crashing", () => {
    const dialog = NoRepositoryDetected()

    expect(dialog).toMatchSnapshot()
  })
})
