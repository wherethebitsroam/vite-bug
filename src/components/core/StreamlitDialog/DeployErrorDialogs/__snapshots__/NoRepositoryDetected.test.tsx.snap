// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeployErrorDialog - NoRepositoryDetected should render without crashing 1`] = `
Object {
  "body": <React.Fragment>
    <p>
      Could not find a remote repository hosted on GitHub. Are you sure you are on a branch that is tracking a remote GitHub branch?
    </p>
    <p>
      How Streamlit Cloud works:
    </p>
    <ul>
      <li>
        To deploy a public app, you must first put it in a public GitHub repo. See
         
        <a
          href="https://docs.streamlit.io"
          rel="noopener noreferrer"
          target="_blank"
        >
          our documentation
        </a>
         
        for more details.
      </li>
      <li>
        If you'd like to deploy a private app,
         
        <a
          href="https://streamlit.io/for-teams"
          rel="noopener noreferrer"
          target="_blank"
        >
          sign up for Streamlit for Teams
        </a>
        .
      </li>
    </ul>
  </React.Fragment>,
  "title": "Unable to deploy app",
}
`;
